src/plugins/hathor_websocket.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { WebSocket } from 'ws';
+import yargs from 'yargs/yargs';
+import { hideBin } from 'yargs/helpers';
+
+import { eventBus } from '../services/notification.service';
+
+/**
+ * @type {WebSocket[]}
+ */
+let sockets = [];
+
+/**
+ * getter for sockets variable.
+ * Used for tests
+ *
+ * @returns {WebSocket[]}
+ */
+export function getSockets() {
+  return sockets;
+}
+
+export const getSettings = () => {
+  const { argv } = yargs(hideBin(process.argv));
+
+  const port = argv.plugin_ws_port || process.env.HEADLESS_PLUGIN_WS_PORT || 8008;
+  return { port };
+};
+
+export function connectionHandler(socket) {
+  sockets.push(socket);
+  console.log('New websocket connection!');
+  socket.on('close', () => {
+    // Remove from connections
+    sockets = sockets.filter(s => s !== socket);
+  });
+}
+
+export function eventHandler(data) {
+  // Broadcast to all live connections
+  for (const socket of sockets) {
+    socket.send(JSON.stringify(data));
+  }
+}
+
+/* istanbul ignore next */
+export const init = async bus => {
```

Comment:
I feel this project needs at least one integration test, and maybe the `websocket` plugin could be well suited for that.

A test suite containing the startup of a wallet, the real-time notification of transactions and a successful shutdown would be great also to serve as an example code to developers.

What do you think?

Reply:
Besides the integration test, I think a comment inside the code pointing to the [`Running the plugins locally`](https://github.com/HathorNetwork/hathor-wallet-headless/pull/237#issuecomment-1253730867) section of your comment would be great. One suggestion of a place to add it would be
https://github.com/HathorNetwork/hathor-wallet-headless/blob/50dd3a3358acd04ab43c2a49194200ba69b6f618/src/controllers/index.controller.js#L182-L183

```
// @see https://github.com/HathorNetwork/hathor-wallet-headless/pull/237#issuecomment-1253730867
```

That, or maybe adding a slightly adapted form of that text to the `readme.md` file under a new _Real-time Notifications_ section.